xbmc/cores/VideoPlayer/VideoPlayer.cpp

```diff
@@ -3132,8 +3144,52 @@ int CVideoPlayer::GetSubtitle()
   return m_SelectionStreams.IndexOf(STREAM_SUBTITLE, *this);
 }
 
+void CVideoPlayer::UpdateStreamInfos()
+{
+  CSingleLock lock(m_player_status_lock);
+  int streamId;
+  std::string retVal;
+
+  // video
+  streamId = GetVideoStream();
```

Comment:
Is there an existing lock for this?
Note this function is called from the main videoplayer thread, where the SelectionsStreams are already being accessed (without an obvious lock I could see). Are you suggesting adding a new lock around every SelectionsStreams access?